entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md

```diff
@@ -47,17 +48,79 @@ EF Core 8 targets .NET 8. Applications targeting older .NET, .NET Core, and .NET
 
 #### Old behavior
 
-Previously, when the `Contains` operator was used in LINQ queries with a parameterized value list, EF generated SQL that was inefficient but worked on all SQL Server versions.
+EF had specialized support for LINQ queries using `Contains` operator over a parameterized value list:
+
+```c#
+var names = new[] { "Blog1", "Blog2" };
+
+var blogs = await context.Blogs
+    .Where(b => names.Contains(b.Name))
+    .ToArrayAsync();
+```
+
+Before EF Core 8.0, EF inserted the parameterized values as constants into the SQL:
+
+```sql
+SELECT [b].[Id], [b].[Name]
+FROM [Blogs] AS [b]
+WHERE [b].[Name] IN (N'Blog1', N'Blog2')
+```
 
 #### New behavior
 
-Starting with EF Core 8.0, EF now generates SQL that is more efficient, but is unsupported on SQL Server 2014 and below.
+Starting with EF Core 8.0, EF now generates SQL that is more efficient in many cases, but is unsupported on SQL Server 2014 and below:
+
+```sql
+SELECT [b].[Id], [b].[Name]
+FROM [Blogs] AS [b]
+WHERE [b].[Name] IN (
+    SELECT [n].[value]
+    FROM OPENJSON(@__names_0) WITH ([value] nvarchar(max) '$') AS [n]
+)
+```
 
 Note that newer SQL Server versions may be configured with an older [compatibility level](/sql/t-sql/statements/alter-database-transact-sql-compatibility-level), also making them incompatible with the new SQL. This can also occur with an Azure SQL database which was migrated from a previous on-premises SQL Server instance, carrying over the old compatibility level.
 
 #### Why
 
-The previous SQL generated by EF Core for `Contains` inserted the parameterized values as constants in the SQL. For example, the following LINQ query:
+The insertion of constant values into the SQL creates many performance problems, defeating query plan caching and causing unneeded evictions of other queries. The new EF Core 8.0 translation uses the SQL Server [`OPENJSON`](/sql/t-sql/functions/openjson-transact-sql) function to instead transfer the values as a JSON array. This solves the performance issues inherent in the previous technique; however, the `OPENJSON` function is unavailable in SQL Server 2014 and below.
+
+For more information about this change, [see this blog post](https://devblogs.microsoft.com/dotnet/announcing-ef8-preview-4/).
+
+#### Mitigations
+
+If your database is SQL Server 2016 (13.x) or newer, or if you're using Azure SQL, check the configured compatibility level of your database via the following command:
+
+```sql
+SELECT name, compatibility_level FROM sys.databases;
```

Comment:
I like that we included how to do this, especially given it isn't always obvious that they are running a different level.